fiftyone/server/query.py

```diff
@@ -222,6 +224,14 @@ def modifier(doc: dict) -> dict:
         doc["evaluations"] = list(doc.get("evaluations", {}).values())
         doc["saved_views"] = doc.get("saved_views", [])
 
+        for d in doc["saved_views"]:
+            # @todo I think the `isinstance(d, dict)` check shouldn't be
+            # required, but currently data with no_dereference=False is getting
+            # here somehow... bug?
+            if isinstance(d, dict):
```

Comment:
@kaixi-wang bringing this to your attention. Without the `isinstance(d, dict)` check, I was getting an error here when launching the App, because a doc with `saved_views` as a list of ObjectIDs was appearing here.

That seems like a bug to me, because that's not what GraphQL expects. And it's not just saved views, runs with ObjectIDs as values were also appearing. IE a non-dereferenced doc is getting here, and I couldn't figure out why at an initial glance.

Reply:
Was the bug consistent? I don't recall ever seeing this bug.

Was there anything unique about the dataset you were loading?
Was this with your change moving the no_dereference to DatasetDocument or before?

Reply:
The bug is consistent. Try loading any dataset with a saved view in the App on this branch with the `if` check commented out.

I don't see anything in this PR that would cause a non-dereferenced doc to get here; I think it's always been a problem but only now raises an error because the reformatting got moved here.

Can you investigate?

Reply:
looked into it and I believe the dereferenced saved_views are coming from the paginator query that directly queries the dataset collection. 
```    data = await foo.aggregate(collection, pipelines)
    results, total = data
    edges = []

    for doc in results:
        _id = doc["_id"]
        edges.append(Edge(node=from_db(doc), cursor=str(_id)))
```
`from_db` calls the Dataset.modifier without calling the `to_dict(no_dereference=True)` serializer, so the saved_views are never resolved from the dbrefs